### DISCUSSION TITLE
ZbStatus3 answer to device MQTT topic

### DISCUSSION THREAD

**Discussion by gonzzzales** [arendst/Tasmota#13041] [Ideas]
Is it possible to make new ZigBee command (ZbStatus4 0xXXXX) that will post information about device NOT to ZbBridge topic (stat/tasmota/RESULT), so to device topic (tele/tasmota/XXXX/SENSOR). This is necessary to get the information about current state of device.

Thanks a lot for the answers

**Comment by barbudor** [comment]
The result of a command should be posted to a topic related to this command.
SENSOR (or STATE) telemetry messages are for recurrent measures.
So a Zigbee sensor like an Aqara temperature sensor will publish automatically to SENSOR

Can you elaborate ?

**Comment by gonzzzales** [comment]
Thanks for the answer. 
I will try to explain in more detail. I have many MQTT devices and have universal structure, each device publish his status in own topic, so i undestand, what device did the message come from. With all types of devices, everything OK, except ZigBee gateway. When system reboot i dont know statuses of devices, connected to ZBBridge, till i got telemetry message. But som devices such as relay or thermostat dont publish it, untill their state changes manually, so this will never happen until i change for example setpoint of thermostat. In this case i want to have a command to check on demand status of device, and this command exists, it is ZBStatus3 0xXXXX. But the problem is, that all answers from all devices come in one topic ( stat/tasmota/RESULT) and i dont undestand, who did the answer come from. Of course i can parse the answer, and check ShortID, but it is not a universal approach for all MQTT devices. So my idea is to make a new command, that work as ZBStatus3, but post the answer in device topic, such as tele/tasmota/XXXX/SENSOR, but i dont know, how to do it, sources of Tasmota are too complicated for me.

**Reply by barbudor** [reply]
So you would like the option for `ZbStatus3` to provide multiple status message.
@s-hadinger will be best to comment on this.

**Comment by rebel7580** [comment]
Perhaps a new setoption like so89 (ZbDeviceTopic) but for ZbStatus3 stat outputs would work?

**Comment by gonzzzales** [comment]
yes, it will be fine

**Comment by s-hadinger** [comment]
I think you get it by the wrong side. The Mqtt gateway should remember the last known state. The reason is that the gateway has much more memory and horsepower than the tiny esp8266. 

ZbStatus onky memorizes a selection of values and gives the last known value. Devices are not actively probed. 

However I can make that change for what it's worth.

**Reply by gonzzzales** [reply]
If it is possible, please make this changes, i think it will be great. For me ZBBridge is a Mqtt gateway, no more devices, which could store values, so it is one way)) 
Thanks a lot for your work!!!
